Fix clang-tidy errors

MatzeB opened the pull request:
Summary: Cleanup clang-tidy complaints in `EmbeddingBag.cpp`: Avoid shadowed variables and unused parameters.

Test Plan: sandcastle

Differential Revision: D71512594

MatzeB commented:
@pytorchbot label "topic: not user facing"

malfet commented:
@pytorchbot merge -i

malfet commented on aten/src/ATen/native/EmbeddingBag.cpp:
```diff
@@ -1065,7 +1065,7 @@ void embedding_bag_cpu_max_out(
     const Tensor& offset2bag,
     const Tensor& output,
-    bool include_last_offset,
+    bool /*include_last_offset*/,
```
`[[maybe_unused]]` is part of C++17 standard, but not associating name with the argument is not part of the standard, so it might not work with VC++
```suggestion
    [[maybe_unused]] bool include_last_offset,
```

MatzeB committed:
Fix clang-tidy errors (#149581)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/149581

Cleanup clang-tidy complaints in `EmbeddingBag.cpp`: Avoid shadowed variables and unused parameters.

Also change code to consistently use `[[maybe_unused]]` annotations.

Test Plan: sandcastle

Reviewed By: malfet

Differential Revision: D71512594

malfet commented:
@pytorchbot merge -i

malfet commented:
@pytorchbot merge -i
